core/src/main/java/org/apache/iceberg/avro/ValueWriters.java

```diff
@@ -238,6 +238,8 @@ public void write(Object s, Encoder encoder) throws IOException {
         encoder.writeString((Utf8) s);
       } else if (s instanceof String) {
         encoder.writeString(new Utf8((String) s));
+      } else if (s instanceof CharSequence) {
+        encoder.writeString((CharSequence) s);
```

Comment:
Why is this needed? We generally want to avoid writing `CharSequence` directly because it will require conversion.